mantidproject/mantid
# Add ability to specify indices of MDHisto to plot

**To test:**
This is what I used for testing, you should try some other data and other plotting functions.
```python
import matplotlib.pyplot as plt
from mantid import plots
from mantid.simpleapi import CreateMDHistoWorkspace
import numpy as np

r=np.linspace(-np.pi, np.pi, 51)
x,y,z = np.meshgrid(r,r,r)
s = np.sin(x) + np.cos(y) + z
ws=CreateMDHistoWorkspace(Dimensionality=3,
                          Extents='-1.02,1.02,-2.04,2.04,-10.2,10.2',
                          SignalInput=s,ErrorInput=s,
                          NumberOfBins='51,51,51',
                          Names='Dim1,Dim2,Dim3',
                          Units='A,B,C')
```

# using indices

```python
#2D plot 
fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
c = ax.pcolormesh(ws, indices=(25,slice(None),slice(None)))
fig.colorbar(c)
plt.show()
```
![Figure_1-13](https://user-images.githubusercontent.com/5595210/56164932-bdc8c100-5f9f-11e9-87f0-b2a77b4b3da7.png)


```python
#1D plot
fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
ax.plot(ws, indices=(0,slice(None),0))
ax.plot(ws, indices=(0,slice(None),26))
ax.plot(ws, indices=(26,slice(None),0))
ax.plot(ws, indices=(26,slice(None),26))
ax.plot(ws, indices=(50,slice(None),0))
ax.legend()
plt.show()
```
![Figure_1-14](https://user-images.githubusercontent.com/5595210/56164945-c1f4de80-5f9f-11e9-88d1-a2bb7a0a2603.png)

# using slicepoint

```python
#2D plot
fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
c = ax.pcolormesh(ws, slicepoint=(0,None,None))
fig.colorbar(c)
plt.show()
```

![Figure_1-15](https://user-images.githubusercontent.com/5595210/56164948-c4efcf00-5f9f-11e9-962c-d8df7f594827.png)

```python
#1D plot
fig, ax = plt.subplots(subplot_kw={'projection':'mantid'})
ax.plot(ws, slicepoint=(0,None,0))
ax.plot(ws, slicepoint=(0,None,1))
ax.plot(ws, slicepoint=(1,None,0))
ax.plot(ws, slicepoint=(1,None,2))
ax.plot(ws, slicepoint=(0,None,4))
ax.legend()
plt.show()
```
![Figure_1-16](https://user-images.githubusercontent.com/5595210/56164953-c7522900-5f9f-11e9-9a6e-95a499c7bb9e.png)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
